lib/Graph/Nodes.cpp

```diff
@@ -591,6 +591,18 @@ void SparseLengthsWeightedSumNode::verify() const {
          "Weights and Indices must have the same size");
 }
 
+void LengthsToRangesNode::verify() const {
+  assert(getResult().getElementType() == getLengths().getElementType() &&
+         "Mismatched element types");
+  assert(getLengths().getElementType() == ElemKind::Int64ITy &&
+         "Lengths must have index type");
```

Comment:
According to the def of Caffe2 op , the input is `"lengths", "1D tensor of int32 segment lengths."`. Just wondering why we need to use "Int64ITy" here?

Reply:
Fair question, I admit I did not really think about the bit width here :-).  IIUC these values are indices, and the only index type we have is Int64ITy (I think Int32QTy is intended for quantized data?).  So I think int64 is safe to use here, if a bit wasteful of memory.

Reply:
Maybe @yinghai could tell us to what extent we should worry about preserving int32 vs int64 indices here.

Reply:
In terms of bit width it's fine. But input tensor from C2 is going to be int32 (https://github.com/pytorch/pytorch/blob/6476e4598ca55626b9d8a28074fa65e10f10f2bb/caffe2/operators/utility_ops.h#L737), hence after ONNX conversion, you are going to get a int32 tensor when you do `onnxSetGraphIO`, so this is an issue?

Reply:
" Int32QTy is intended for quantized data ? " @rdzhabarov could you please verify it? I know if we create a type with "Int32QTy", the scale and offset should be provided. But I think we could just provide some dummy scale and offset for a regular int32_t :P If not, we might need to added some thing for the int32_t type.

Reply:
Hah, there is actually an issue because Glow importer doesn't support `int32` yet: https://github.com/pytorch/glow/blob/777f01c7f25fa88f77462b12904b0e89c29d84b9/lib/Importer/ONNXIFIModelLoader.cpp#L26-L40

Well, at C2 side, we didn't handle integer tensor either: https://github.com/pytorch/pytorch/blob/master/caffe2/opt/onnxifi_transformer.cc#L80-L81. So there is some work to do. :)

Reply:
@beicy Yeah we technically could do something hacky like that with `Int32QTy`, but if we want to use the data as indices we really should add `Int32ITy`.

Reply:
> " Int32QTy is intended for quantized data ? "

True

I agree with Jordan.

Reply:
In that case, I will add "Int32ITy" for this one and RowwiseQuantizedFC and "UInt8QTy" for quantized uint8_t activations.